data-app/src/enhance/courseInfo.py

```diff
@@ -29,53 +30,82 @@ def _calculate_current_term(self) -> str:
+                        
+                        libed_dist = session.query(Libed).filter(Libed.name == libed_name).first()
+                        if libed_dist and class_dist not in libed_dist.class_dists:
+                            libed_dist.class_dists.append(class_dist)
+                        
+                    print(f"[CI Enhance] Updated [{class_dist.campus}] {class_dist.dept_abbr} {class_dist.course_num} : Libeds: ({class_dist.libeds})")
+            session.commit()
+        except Exception as e:
+            print(f"[CI ERROR] Error processing course data for {dept} at {campus}: {e}")
+            session.rollback()
+        finally:
+            session.close()
+
+    async def enhance_helper(self, dept_dist: DepartmentDistribution, semaphore: asyncio.Semaphore) -> None:
+
+        async with semaphore:
+            dept = dept_dist.dept_abbr
+            campus = dept_dist.campus
+            campus_str = str(campus)
+
+            # Only process UMNTC and UMNRO campuses
+            if campus_str not in ["UMNTC", "UMNRO"]:
+                return
+            
+            current_term = self._calculate_current_term()
+            link = f"https://courses.umn.edu/campuses/{campus_str.lower()}/terms/{current_term}/courses.json?q=subject_id={dept}"
 
-        # Only process UMNTC and UMNRO campuses
-        if campus_str not in ["UMNTC", "UMNRO"]:
-            return
+            courses = []
+            async with httpx.AsyncClient() as client:
+                try:
+                    response = await client.get(link)
+                    response.raise_for_status()
+                    req = response.json()
+                    courses = req.get("courses", [])
+                except httpx.HTTPStatusError as e:
+                    print(f"[CI HTTP ERROR] Failed for {dept}. Status: {e.response.status_code}. URL: {e.request.url}")
+                    return
+                except json.JSONDecodeError as e:
+                    print(f"[CI PARSE ERROR] Failed to parse JSON for {dept}. URL: {link}. Error: {repr(e)}")
+                    return
+                except httpx.RequestError as e:
+                    print(f"[CI REQUEST ERROR] Failed to fetch data for {dept}. Error: {repr(e)}") 
+                    return
         
-        current_term = self._calculate_current_term()
-        link = f"https://courses.umn.edu/campuses/{campus_str.lower()}/terms/{current_term}/courses.json?q=subject_id={dept}"
+            if not courses:
+                print(f"[CI INFO] No courses found for {dept} at {campus_str}")
+                return
 
-        with requests.get(link) as url:
             try:
-                req = url.json()
-                courses = req.get("courses", [])
-            except ValueError:
-                print("Json malformed, icky!")
+                await asyncio.to_thread(self._process_course_data, courses, dept, campus)
```

Comment:
This spawns a seperate thread to run a synchronous function. Can you make `_process_course_data` async and just await it instead? Would reduce thread overhead.